manifests/plugin/discovery.pp

```diff
@@ -1,16 +1,18 @@
-# @summary Install the discovery plugin and images on smart proxies
+# = Foreman Proxy Discovery plugin
 #
-# @param install_images
-#   should the discovery image be downloaded and extracted
+# Install the discovery plugin and images on smart proxies
 #
-# @param tftp_root
-#   tftp root to install image into
+# === Parameters:
 #
-# @param source_url
-#   source URL to download from
+# $install_images::   should the discovery image be downloaded and extracted
 #
-# @param image_name
-#   tarball with images
+# === Advanced parameters:
+#
+# $tftp_root::   tftp root to install image into
+#
+# $source_url::   source URL to download from
```

Comment:
I disagree. For example, you mirror it somewhere and only change the base URL while still using the same imagine name as upstream. Another example could be branding where you change the source URL to your company name. It also makes the code a lot harder because now you need to split some URL. That's always painful and a good source of bugs. Given these are advanced parameters, I wouldn't change this.